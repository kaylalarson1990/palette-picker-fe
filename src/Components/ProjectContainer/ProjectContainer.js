import React from "react";
import { connect } from "react-redux";
import Projects from "../Projects/Projects";

export const ProjectContainer = props => {
  const savedProjects = props.projects.length ?
        props.projects.reverse().map(project => (
          <Projects key={project.id} project={project} />
        ))
        : <p>Start saving your favorite color palettes to a new project!</p>

  return (
    <section class="saved-projects-section">
      {savedProjects}
    </section>
  );
};

export const mapStateToProps = state => ({
  projects: state.projects
});

export default connect(mapStateToProps)(ProjectContainer);