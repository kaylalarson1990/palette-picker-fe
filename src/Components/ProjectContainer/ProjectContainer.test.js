import React from "react";
import { ProjectContainer } from "./ProjectContainer";
import { shallow } from "enzyme";
import "../../../setupTests";

describe("ProjectContainer", () => {
  let wrapper, instance, mockProject;
  beforeEach(() => {
    mockProject = [
      {
        id: 13,
        project_name: "Test Project"
      }
    ];
    wrapper = shallow(<ProjectContainer projects={mockProject} />);
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
