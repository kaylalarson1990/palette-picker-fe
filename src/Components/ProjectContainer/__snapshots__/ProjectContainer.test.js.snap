// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectContainer should match the snapshot 1`] = `
<Fragment>
  <section
    className="project-container"
  >
    <Connect(NewProjectForm) />
  </section>
  <section
    className="saved-projects-section"
  >
    <Projects
      key="13"
      name="Test Project"
    />
  </section>
</Fragment>
`;
